app/validation/validator.py

```diff
@@ -240,6 +240,14 @@ def validate_date_range(self, question):
                 errors.append(self._error_message('Days can not be used in period_limit for yyyy-mm date range for {}'
                                                   .format(question['id'])))
 
+            if (first_answer_type == 'YearDate'
+                    and ('minimum' in period_limits and ('days' in period_limits['minimum']
+                                                         or 'months' in period_limits['minimum'])
+                         or 'maximum' in period_limits and ('days' in period_limits['maximum']
+                                                            or 'months' in period_limits['maximum']))):
+                errors.append(self._error_message('Days/Months can not be used in period_limit for yyyy date range'
+                                                  ' for {}'.format(question['id'])))
+
```

Comment:
Also `R:243,16: Too many boolean expressions in if statement (7/5) (too-many-boolean-expressions)`

Reply:
I can split this into two separate statements. One for Days and one for Months but that looks messy and could return twice as many errors as necessary. I'll see if I can reduce the boolean expressions but keep it simple to understand. I didn't want a complex "If" that's actually harder to understand but has less expressions.

Reply:
Possible suggestion:

```
if first_answer_type == 'YearDate':
    has_min_limit = 'days' in period_limits['minimum'] or 'months' in period_limits['minimum']
    has_max_limit = 'days' in period_limits['maximum'] or 'months' in period_limits['maximum']
    if has_min_limit or has_max_limit:
        errors.append(self._error_message('Days/Months can not be used in period_limit for yyyy date range'
                                        ' for {}'.format(question['id'])))
```

Reply:
As above but:

Reply:
This can actually then go above and be used for mm-yyyy type as well.

Reply:
Nice 👏

Reply:
Bravo 👍